### ISSUE TITLE
add nullability to stringForColumn and other methods

### ISSUE THREAD

**Issue by johnryan** [ccgus/fmdb#516]
Currently if you save a NSNull value into a string column in swift then read it out it will cause a crash. i.e.

`   let name = results.stringForColumn("name")

because without nullability the swift code generated looks like:

    public func stringForColumn(columnName: String!) -> String!

Adding nullability on these functions should prevent the crash.

    - (NSString*)stringForColumn:(NSString*)columnName;

should become:

    - (nullable NSString*)stringForColumn:(NSString*)columnName;

**Comment by robertmryan** [comment]
I agree that FMDB should be audited for nullability, but I would advise that this be deferred until the next major release in the code. This isn't something to be tackled in a minor release because this can cause non-backward compatible changes for anyone who wrote code against the non-audited version (i.e. anyone who wrote code dependent upon the behavior of implicitly unwrapped optionals). We don't want to be issuing non-backward compatible changes during minor releases of FMDB.

In terms of your crash, I suspect this is happening because you are trying to carry on using the implicitly unwrapped `name`, without actually checking to see if it is `nil` or not. I would advise that you treat `name` like it was an optional value it is, e.g.:

```swift
if let name = results.stringForColumnName("name") { ... }
```

Or 

```
guard let name = results.stringForColumnName("name") else { ... }
```

Note, this is the same code that you'd write if (and when) `stringForColumn` is correctly designated as `nullable`. So, go ahead and adopt this conditional binding syntax, and you'll be all prepared for if and when FMDB is audited for nullability.

**Comment by jeffbailey** [comment]
Glad to see this was added to the 3.0 milestone.  Having methods like stringForColumnName return an optional instead of an implicitly unwrapped optional will lead to much safer code.  Thanks!
